pkg/lib/config/ui.go

```diff
@@ -43,6 +44,14 @@ type UIConfig struct {
 	Implementation UIImplementation `json:"implementation,omitempty"`
 	// ForgotPassword is the config for the default auth ui
 	ForgotPassword *UIForgotPasswordConfig `json:"forgot_password,omitempty"`
+	// AuthenticationFlow configures ui behaviour of authentication flow under default auth ui
+	AuthenticationFlow *UIAuthenticationFlowConfig `json:"authentication_flow,omitempty"`
+}
+
+func (c *UIConfig) SetDefaults() {
+	if c.AuthenticationFlow == nil {
+		c.AuthenticationFlow = &UIAuthenticationFlowConfig{}
+	}
```

Comment:
SetDefaults is only necessary when you want to assign values to strings, ints, or booleans. Pointer to struct will be populated by `PopulateDefaultValues`. So remove this SetDefaults()